tools/clang/lib/SPIRV/SpirvBasicBlock.cpp

```diff
@@ -20,5 +20,19 @@ bool SpirvBasicBlock::hasTerminator() const {
   return !instructions.empty() && isa<SpirvTerminator>(instructions.back());
 }
 
+bool SpirvBasicBlock::invokeVisitor(Visitor *visitor) {
+  if (!visitor->visit(this, Visitor::Phase::Init))
+    return false;
+
+  for (auto inst : instructions)
```

Comment:
I'd suggest to use `auto*` here to make it clear and also guard against accidentally changing instructions from pointers to objects.